Source/Session/Session.swift

```diff
@@ -310,6 +310,13 @@ extension Session: WebViewDelegate {
         delegate?.session(self, didProposeVisit: proposal)
     }
     
+    func webView(_ bridge: WebViewBridge, didProposeVisitToCrossOriginRedirect location: URL) {
+        // Remove the current visitable from the backstack since it
+        // resulted in a visit failure due to a cross-origin redirect.
+        activatedVisitable?.visitableViewController.navigationController?.popViewController(animated: false)
```

Comment:
Is this the best way to handle popping the current `visitable` ViewController from the backstack?

Reply:
Could this be calling a delegate instead? This would allow customization of the behavior here and the `Session` would be less dependent on the app being structured in a specific way, ie. the `Session` doesn't really 'push' or 'pop' any controllers today - controllers just need to implement `Visitable`.

Reply:
Yeah, I figured that'd probably be where we ended up, makes sense 👍

Reply:
An alternative would be to implement a method in `Visitable` that provides the behavior as in the PR, but could be overwritten and customized in the controller that implements `Visitable`, ie for. controllers not living in a `UINavigationController`, eg.:

```swift
// Session
func webView(_ bridge: WebViewBridge, didProposeVisitToCrossOriginRedirect location: URL) {
   activatedVisitable?.removeDueToCrossOriginRedirect()
}

// Visitable
func removeDueToCrossOriginRedirect() {
  // Sensible default, but could be customized
  visitableViewController.navigationController?.popViewController(animated: false)
}
```

Reply:
Would it be better to perform the visit with a REPLACE action instead of manually manipulating the backstack?

Reply:
Ah, gotcha. Then I think we should find a way to delegate to the navigator to perform the stack manipulation.

Reply:
After digging into the PR, I'm leaning more towards @pfeiffer's recommendation of an additional delegate callback on `SessionDelegate`. This would require the developer to do the popping manually but we can handle that for them in Turbo Navigator. I like this approach because if the developer isn't using a navigation controller then they have full control of what to do.

But more importantly, if the redirected screen was opened in a modal then we need to _dismiss_ not _pop_ the new (blank) view controller from the screen. This will be a one-liner in Turbo Navigator because we already have logic that handles those situations.